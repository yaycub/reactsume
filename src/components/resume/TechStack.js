import React from 'react';
import PropTypes from 'prop-types';
import TechStackItem from './TechStackItem';

const TechStack = ({ languages, frontEnd, backEnd, dataBases }) => {
  const languageEls = techStackElCreator(languages);
  const frontEndEls = techStackElCreator(frontEnd);
  const backEndEls = techStackElCreator(backEnd);
  const dataBasesEls = techStackElCreator(dataBases);

  return (
    <ul>
      Languages:
      {languageEls}
      Front-end:
      {frontEndEls}
      Back-end:
      {backEndEls}
      Databases:
      {dataBasesEls}
    </ul>
  );
};

TechStack.propTypes = {
  languages: PropTypes.array.isRequired,
  frontEnd: PropTypes.array.isRequired,
  backEnd: PropTypes.array.isRequired,
  dataBases: PropTypes.array.isRequired
};

function techStackElCreator(arr){
  return arr.map((item, i) => {
    return (
      <li key={`${i}+${item}`}>
        <TechStackItem item={item} />
      </li>
    );
  });
}

export default TechStack;
