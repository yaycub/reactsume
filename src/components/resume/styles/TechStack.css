.TechStack{
  display: flex;
  flex-direction: column;

  h3{
    font-size: 15px;
    font-weight: bold;
    margin-top: 5px;
    margin-bottom: 10px;
  }

  h4{
    font-size: 13px;
    margin-top: 5px;
    margin-bottom: 5px;
    font-weight: bold;
  }

  li{
    list-style: square;
    font-size: 12px;
    margin-left: 40px;
    margin-top: 2px;
    margin-bottom: 2px;
    font-style: italic;
  }
}