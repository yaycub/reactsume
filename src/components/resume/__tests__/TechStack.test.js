import React from 'react';
import { shallow } from 'enzyme';
import data from '../../../resume.json';
import TechStack from '../TechStack.js';

describe('TechStack component', () => {
  it('can render a TechStack', () => {
    const { techStack } = data;
    const wrapper = shallow(<TechStack {...techStack} />);
    expect(wrapper).toMatchSnapshot();
  });
});
