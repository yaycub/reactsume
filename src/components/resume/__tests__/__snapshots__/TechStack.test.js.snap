// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TechStack component can render a TechStack 1`] = `
<ul>
  <h3>
    Tech Stack
  </h3>
  <h4>
    Languages:
  </h4>
  <li
    key="0+HTML/CSS"
  >
    <TechStackItem
      item="HTML/CSS"
    />
  </li>
  <li
    key="1+JavaScript"
  >
    <TechStackItem
      item="JavaScript"
    />
  </li>
  <li
    key="2+SQL"
  >
    <TechStackItem
      item="SQL"
    />
  </li>
  <h4>
    Front-end:
  </h4>
  <li
    key="0+React.js"
  >
    <TechStackItem
      item="React.js"
    />
  </li>
  <li
    key="1+Redux"
  >
    <TechStackItem
      item="Redux"
    />
  </li>
  <h4>
    Back-end:
  </h4>
  <li
    key="0+Node.js"
  >
    <TechStackItem
      item="Node.js"
    />
  </li>
  <li
    key="1+Express"
  >
    <TechStackItem
      item="Express"
    />
  </li>
  <li
    key="2+Mongoose"
  >
    <TechStackItem
      item="Mongoose"
    />
  </li>
  <h4>
    Databases:
  </h4>
  <li
    key="0+PostgreSQL"
  >
    <TechStackItem
      item="PostgreSQL"
    />
  </li>
  <li
    key="1+MongoDB"
  >
    <TechStackItem
      item="MongoDB"
    />
  </li>
</ul>
`;
